### PR TITLE
feat: Config diff phase 2

### PR DESCRIPTION
**Pull request by RohitRaj011**
# Description

chore: Move `Collapsible List Component` from 'dashboard' to 'common-lib'
chore: Add borderless variant to `SelectPicker`
feat: DeploymentConfigDiff Component
feat: Add sorting support to `DeploymentHistoryDiffView`

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [x] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiffMain.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    isLoading,
+    headerText = 'Compare With',
+    configList = [],
+    scrollIntoViewId,
+    selectorsConfig,
+    sortOrder,
+    onSortBtnClick,
+}: DeploymentConfigDiffMainProps) => {
+    // STATES
+    const [expandedView, setExpandedView] = useState<Record<string | number, boolean>>({})
```
Comment (eshankvaish) [inline src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiffMain.tsx:23]: This state exists in DeploymentConfigDiffNavigation as well. Pls check if redundant

File: src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiffMain.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            const { heading: primaryHeading, list: primaryList } = primaryConfig
+            const { heading: secondaryHeading, list: secondaryList } = secondaryConfig
+
+            return (
+                <DeploymentConfigDiffAccordion
+                    key={`${id}-${title}`}
+                    ref={scrollIntoViewId === id ? scrollRef : null}
+                    id={id}
+                    title={title}
+                    isExpanded={!expandedView[id]}
```
Comment (eshankvaish) [inline src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiffMain.tsx:93]: Please confirm if this check is correct.
```suggestion
                    isExpanded={expandedView[id]}
```

File: src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiffAccordion.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+export const DeploymentConfigDiffAccordion = forwardRef<HTMLDivElement, DeploymentConfigDiffAccordionProps>(
+    ({ hasDiff, children, title, id, isExpanded, handleOnClick }, ref) => (
+        <div ref={ref} id={id} className="dc__border br-4 deployment-config-diff__accordion">
+            <button
+                type="button"
+                className="dc__unset-button-styles px-16 py-10 flexbox dc__align-items-center dc__gap-8 w-100 br-4 bcn-50 dc__position-sticky dc__top-12 dc__zi-10"
+                style={{
+                    top: 0,
+                }}
```
Comment (eshankvaish) [inline src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiffAccordion.tsx:15]: do we have a util class for this?

File: src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiff.scss [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+        &__header:last-child {
+            flex: 1 0 50%;
+        }
+    }
+
+    &__main-content {
+        height: calc(100vh - 122px);
+        overflow: auto;
+    }
```
Comment (eshankvaish) [inline src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiff.scss:25]: Suggestion: it would be great if we can get rid of calc here

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiffMain.tsx (hunk 1 above)
Comment (RohitRaj011) [reply src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiffMain.tsx:23]: 2 different states
one for side nav collapsible list and one for accordion in main content

File: src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiff.scss (hunk 4 above)
Comment (RohitRaj011) [reply src/Shared/Components/DeploymentConfigDiff/DeploymentConfigDiff.scss:23]: it is required

File: src/Shared/Components/SelectPicker/utils.ts [hunk 5]
```diff
@@ -14,26 +14,52 @@ const getMenuWidthFromSize = (menuSize: SelectPickerProps['menuSize']): string =
...
+    menu: (base, state) => ({
         ...base,
         overflow: 'hidden',
         marginBlock: '4px',
         backgroundColor: 'var(--N0)',
         border: '1px solid var(--N200)',
         boxShadow: '0px 2px 4px 0px rgba(0, 0, 0, 0.20)',
         width: getMenuWidthFromSize(menuSize),
+        zIndex: 'var(--select-picker-menu-index)',
+        ...(getVariantOverrides(variant)?.menu(base, state) || {}),
```
Comment (eshankvaish) [inline src/Shared/Components/SelectPicker/utils.ts:62]: this would still break i guess, please verify once

alt text

**Review by eshankvaish — APPROVED** [review]

File: src/Shared/Components/CICDHistory/DeploymentHistoryDiff/DeploymentHistoryDiffView.tsx [hunk 6]
```diff
@@ -77,12 +79,16 @@ const DeploymentHistoryDiffView = ({
             value={
                 !baseTemplateConfiguration?.codeEditorValue?.value || isDeleteDraft
                     ? ''
-                    : YAMLStringify(JSON.parse(editorValuesRHS))
+                    : YAMLStringify(JSON.parse(editorValuesRHS), {
```
Comment (vivek-devtron) [inline src/Shared/Components/CICDHistory/DeploymentHistoryDiff/DeploymentHistoryDiffView.tsx:82]: Extract all this logic out

**Pull request merged**